apis/risingwave/v1alpha1/risingwave_state_store_backend.go

```diff
@@ -0,0 +1,276 @@
+	RisingWaveMinIOCredentials `json:"credentials"`
+
+	// Secret contains the credentials to access the MinIO service. It must contain the following keys:
+	//   * username
+	//   * password
+	// +kubebuilder:validation:Required
+	Secret string `json:"secret"`
+
+	// Endpoint of the MinIO service.
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+
+	// Bucket of the MinIO service.
+	// +kubebuilder:validation:Required
+	Bucket string `json:"bucket"`
+}
+
+// RisingWaveS3Credentials is the reference and keys selector to the AWS access credentials stored in a local secret.
+type RisingWaveS3Credentials struct {
+	// The name of the secret in the pod's namespace to select from.
+	SecretName string `json:"secretName"`
+
+	// AccessKeyRef is the key of the secret to be the access key. Must be a valid secret key.
+	// Defaults to "AccessKeyID".
+	// +kubebuilder:default=AccessKeyID
+	AccessKeyRef *string `json:"accessKeyRef,omitempty"`
+
+	// SecretAccessKeyRef is the key of the secret to be the secret access key. Must be a valid secret key.
+	// Defaults to "SecretAccessKey".
+	// +kubebuilder:default=SecretAccessKey
+	SecretAccessKeyRef *string `json:"secretAccessKeyRef,omitempty"`
+}
+
+// RisingWaveStateStoreBackendS3 is the collection of parameters for the S3 backend state store.
+type RisingWaveStateStoreBackendS3 struct {
+	// RisingWaveS3Credentials is the credentials provider from a Secret.
+	RisingWaveS3Credentials `json:"credentials"`
+
+	// Secret contains the credentials to access the AWS S3 service. It must contain the following keys:
+	//   * AccessKeyID
+	//   * SecretAccessKey
+	//   * Region (optional if region is specified in the field.)
+	// +kubebuilder:validation:Required
+	Secret string `json:"secret"`
+
+	// Bucket of the AWS S3 service.
+	// +kubebuilder:validation:Required
+	Bucket string `json:"bucket"`
+
+	// Region of AWS S3 service. It is an optional field that overrides the `Region` key from the secret.
+	// Specifying the region here makes a guarantee that it won't be changed anymore.
+	Region string `json:"region,omitempty"`
+
+	// Endpoint of the AWS (or other vendor's S3-compatible) service. Leave it empty when using AWS S3 service.
+	// You can reference the `REGION` and `BUCKET` variables in the endpoint with `${BUCKET}` and `${REGION}`, e.g.,
+	//   s3.${REGION}.amazonaws.com
+	//   ${BUCKET}.s3.${REGION}.amazonaws.com
+	// +optional
+	// +kubebuilder:validation:Pattern="^(?:https://)?(?:[^/.\\s]+\\.)*(?:[^/\\s]+)*$"
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// VirtualHostedStyle indicates to use a virtual hosted endpoint when endpoint is specified. The operator automatically
+	// adds the bucket prefix for you if this is enabled. Be careful about doubly using the style by specifying an endpoint
+	// of virtual hosted style as well as enabling this.
+	VirtualHostedStyle bool `json:"virtualHostedStyle,omitempty"`
```

Comment:
I think we can remove this field. Users can specify an endpoint in virtual hosted style on their own. Removing this field may require a conversion by setting the endpoint to the valid one, e.g.,

```yaml
# legacy
endpoint: https://s3.${REGION}.amazonaws.com
virtualHostedStyle: true

# new
endpoint: https://${BUCKET}.s3.${REGION}.amazonaws.com
```